arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/signals/AbortingConditions.java

```diff
@@ -30,72 +33,79 @@
 public final class AbortingConditions {
   private boolean callStackOverflow;
   private boolean insufficientBalance;
+  private boolean maxNonceReached;
 
   /**
    * @param callStackOverflow too many nested contexts
    * @param insufficientBalance trying to give more ETH than the caller has
+   * @param maxNonceReached the nonce of the creator account is too high (EIP-2681)
    */
-  public AbortingConditions(boolean callStackOverflow, boolean insufficientBalance) {
+  public AbortingConditions(
+      boolean callStackOverflow, boolean insufficientBalance, boolean maxNonceReached) {
     this.callStackOverflow = callStackOverflow;
     this.insufficientBalance = insufficientBalance;
-  }
-
-  public static AbortingConditions of(Hub hub) {
-    AbortingConditions r = new AbortingConditions();
-    r.prepare(hub);
-    return r;
+    this.maxNonceReached = maxNonceReached;
   }
 
   public void reset() {
-    this.callStackOverflow = false;
-    this.insufficientBalance = false;
+    callStackOverflow = false;
+    insufficientBalance = false;
+    maxNonceReached = false;
   }
 
   public void prepare(Hub hub) {
-    this.callStackOverflow = hub.callStack().wouldOverflow();
+    callStackOverflow = hub.callStack().wouldOverflow();
     if (this.callStackOverflow) {
       return;
     }
```

Comment:
Are we 100% sure that we don't want the full picture of aborting conditions ? AFAIR returning after detecting 1 aborting condition should be ok. IIRC all aborting conditions are detected in one fell swoop by the OOB module so this should be ok but I don't remember in detail.

Reply:
As the OOB does the three aborting in the same call, it should be ok ... It was already implemented like this when we had only 2 aborting conditions